pkg/local_object_storage/metabase/metadata.go

```diff
@@ -0,0 +1,237 @@
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	objectcore "github.com/nspcc-dev/neofs-node/pkg/core/object"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+	"github.com/nspcc-dev/neofs-sdk-go/version"
+	"go.etcd.io/bbolt"
+)
+
+const (
+	metaPrefixA  = byte(iota)
+	metaPrefixBI // integer attributes
+	metaPrefixBS // all other attributes
+	metaPrefixC
+)
+
+var (
+	maxUint256 = new(big.Int).SetBytes([]byte{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+		255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255})
+	maxUint256Neg = new(big.Int).Neg(maxUint256)
+)
+
+// TODO: fill on Init
+// TODO: system attributes
+func putMetadata(tx *bbolt.Tx, cnr cid.ID, id oid.ID, ver version.Version, owner user.ID, typ object.Type, creationEpoch uint64,
+	payloadLen uint64, pldHash, pldHmmHash, splitID []byte, parentID, firstID oid.ID, attrs []object.Attribute) error {
+	mb, err := tx.CreateBucketIfNotExists(metaBucketKey(cnr))
+	if err != nil {
+		return fmt.Errorf("create meta bucket for container: %w", err)
+	}
+	idk := [1 + oid.Size]byte{metaPrefixA}
+	copy(idk[1:], id[:])
+	if err := mb.Put(idk[:], nil); err != nil {
+		return fmt.Errorf("put object ID to container's meta bucket: %w", err)
+	}
+
+	var k []byte
+	// TODO: move to global funcs
+	makeKeyB := func(prefix byte, attr string, valLen int) (int, int) {
+		ln := 1 + oid.Size + len(attr) + valLen + len(utf8Delimiter)*2 // TODO: constantize some stuff
+		if len(k) < ln {
+			k = make([]byte, ln)
+		}
+		k[0] = prefix
+		off := 1 + copy(k[1:], attr)
+		off += copy(k[off:], utf8Delimiter)
+		valOff := off
+		off += valLen
+		off += copy(k[off:], utf8Delimiter)
+		copy(k[off:], id[:])
+		return ln, valOff
+	}
+	makeKeyC := func(attr string, valLen int) (int, int) {
+		ln := 1 + oid.Size + len(attr) + valLen + len(utf8Delimiter) // TODO: constantize some stuff
+		if len(k) < ln {
+			k = make([]byte, ln)
+		}
+		k[0] = metaPrefixC
+		off := 1 + copy(k[1:], id[:])
+		off += copy(k[off:], attr)
+		off += copy(k[off:], utf8Delimiter)
+		return ln, off
+	}
+	// TODO: make generic to pass []byte directly
+	putPlain := func(attr, val string) error {
+		kLn, vOff := makeKeyB(metaPrefixBS, attr, len(val))
+		copy(k[vOff:], val)
+		if err := mb.Put(k[:kLn], nil); err != nil {
+			return fmt.Errorf("put object attribute %q to container's meta bucket: %w", attr, err)
+		}
+		kLn, vOff = makeKeyC(attr, len(val))
+		copy(k[vOff:], val)
+		if err := mb.Put(k[:kLn], nil); err != nil {
+			return fmt.Errorf("put object attribute %q to container's meta bucket: %w", attr, err) // TODO: distinguishable context
+		}
+		return nil
+	}
+	putInt := func(attr string, n *big.Int) error {
+		// TODO: check uint256 overflow
+		kLn, vOff := makeKeyB(metaPrefixBI, attr, 1+32) // sign + 256-bit. Sign makes some sensible order
```

Comment:
I see you're using `sign+value` encoding here (not 256 fixed), this is related to 256 signed note. Let's think a bit about. A separate byte can be easier to handle instead of sign bit, but some numbers can be incompatible with VM. Does it matter? Not sure.